- Первое задание

AlexSunset opened the pull request.

Commit:
- Первое задание

pomegranate282922 commented on pom.xml:
```diff
@@ -9,8 +9,73 @@
+            <version>4.13.2</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
```
Не нужно использовать junit4 и junit5 в одном проекте.
Нужно исправить

pomegranate282922 commented on src/test/java/BunTest.java:
```diff
@@ -0,0 +1,20 @@
+    public void getPriceReturnThisPrice(){
+        Assert.assertEquals(bun.price, bun.getPrice(), 0);
+    }
+
```
Здесь ожидается использование параметризации и проверка разных классов эквивалентности 
Нужно исправить

pomegranate282922 commented on src/test/java/BurgerParameterTest.java:
```diff
@@ -0,0 +1,47 @@
+                        Stream.of(new Ingredient(IngredientType.SAUCE, "Ing1", 20),
+                                  new Ingredient(IngredientType.FILLING, "Ing2", 30))
+                                .collect(Collectors.toList()), 110},
+        };
```
Отлично, что используется параметризация. Но не очень понятен выбор тестовых данных. Вроде бы они все из одного класса эквивалентности. Зачем параметризация?

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,75 @@
+        MatcherAssert.assertThat(burger.getReceipt(), containsString("= filling TestFilling ="));
+        MatcherAssert.assertThat(burger.getReceipt(), containsString("Price: 130,000000"));
+    }
+}
```
Ожидается, что в тестах на бургер для всех(!)  зависимостей будут использованы моки.
Нужно исправить.

pomegranate282922 commented on src/test/java/IngredientTypeTest.java:
```diff
@@ -0,0 +1,11 @@
+public class IngredientTypeTest {
+
+    @Test
+    public void IngredientTypeHasTwoElements(){
```
Методы именуем начиная со строчной буквы. Нужно исправить

pomegranate282922 reviewed (commented):
Работа хорошая! Но есть и важные замечания - использование моков, параметризации, нейминг, остальное по мелочи.

Commit:
- Исправления по замечаниям
- Не понял про параметризацию в BunTest. Это же тест на геттер с float значением. Поиск классов эквивалентности привёл меня в явное замешательство. Я сделал просто по букве написанной, но моё естесство противистя этому. Прошу пояснить

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,97 @@
+    public void removeIngredientRemoveItemFromIngredientList() {
+        assertEquals(0, burger.ingredients.size());
+        burger.ingredients.add(ingredient);
+        Assert.assertTrue(burger.ingredients.contains(ingredient));
```
Зачем этот ассерт тут? Так ты нарушаешь принцип: один тест - одна проверка.
Проверяем то, для чего написан тест
Нужно исправить везде

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,97 @@
+        assertEquals(0, burger.ingredients.get(0).price, 0);
+        burger.moveIngredient(2, 0);
+        assertEquals(3, burger.ingredients.get(0).getPrice(), 0);
+    }
```
Здесь то же самое - должна быть одна проверка одного метода.

pomegranate282922 reviewed (commented):
Многое поправил, но ещё хочется увидеть настоящие юнит тесты.

Commit:
- Убрал лишние assert

pomegranate282922 reviewed (approved):
Ок, зачёт!
